src/phoenix/server/templates/index.html

```diff
@@ -1,3 +1,43 @@
+{% set rendered_files = [] %}
+{% set rendered_chunks = [] %}
+
+{% macro render_chunk(chunk_name, level=0) %}
+  {% if chunk_name not in rendered_chunks %}
+    {% set _ = rendered_chunks.append(chunk_name) %}
+    {% set chunk = manifest[chunk_name] %}
+    
+    {% if chunk.css %}
+      {% for css_file in chunk.css %}
+        {% if css_file not in rendered_files %}
+          <link rel="stylesheet" href="{{ basename }}/{{ css_file }}">
+          {% set _ = rendered_files.append(css_file) %}
+        {% endif %}
+      {% endfor %}
+    {% endif %}
+    
+    {% if chunk.imports %}
+      {% for import in chunk.imports %}
+        {{ render_chunk(import, level + 1) }}
+      {% endfor %}
+    {% endif %}
+    
+    {% if level == 0 %}
+      {% if chunk.file.endswith('.js') and chunk.file not in rendered_files %}
+        <script type="module" src="{{ basename }}/{{ chunk.file }}"></script>
+        {% set _ = rendered_files.append(chunk.file) %}
+      {% elif chunk.file.endswith('.css') and chunk.file not in rendered_files %}
+        <link rel="stylesheet" href="{{ basename }}/{{ chunk.file }}">
+        {% set _ = rendered_files.append(chunk.file) %}
+      {% endif %}
+    {% endif %}
+    
+    {% if chunk.file.endswith('.js') and chunk.file not in rendered_files %}
+      <link rel="modulepreload" href="{{ basename }}/{{ chunk.file }}">
+      {% set _ = rendered_files.append(chunk.file) %}
+    {% endif %}
+  {% endif %}
+{% endmacro %}
+
```

Comment:
there's a bunch of whitespace here. It's not a huge deal but kinda confusing when inspecting the page.

Reply:
fixed!